tfhe/src/shortint/server_key/tests/shortint.rs

```diff
@@ -2298,21 +2360,13 @@ where
         let ctxt_1 = cks.encrypt(clear_1);
 
         // add the two ciphertexts
-        let mut ct_res = sks.mul_lsb(&ctxt_0, &ctxt_1);
+        let ct_res = sks.mul_lsb(&ctxt_0, &ctxt_1);
 
-        let mut clear = clear_0 * clear_1;
+        let clear = clear_0 * clear_1;
 
-        //add multiple times to raise the degree
-        for _ in 0..30 {
-            ct_res = sks.mul_lsb(&ct_res, &ctxt_0);
-            clear = (clear * clear_0) % modulus;
-
-            // decryption of ct_res
-            let dec_res = cks.decrypt(&ct_res);
+        let dec_res = cks.decrypt(&ct_res);
 
-            // assert
-            assert_eq!(clear, dec_res);
```

Comment:
We don't need this step in default operation, since there is a full-propagate at the beginning and the end of the operation.
This must be a copy-paste error while creating the test.

Reply:
default tests have  two things to test in order to go into all branches possible

- test the operation when inputs are clear of carries
- test the operation when inputs do have carries

The second inner loop that was removed tested the first case (input with clean carries
which the new test still test.

 The comment `//add multiple times to raise the degree` was wrong, their was no degree being raised because it used the default ops

Reply:
The difference is that instead of testing 10*30 ops we are now testing NB_TEST ops

Reply:
wasn't the inner loop for carry stuff ? In any case I was surprised if it was meant for coverage only

Reply:
The comment says the inner loop is for raising degree, testing case where carries are non empty.

However the code uses `mul_lsb` in that case (in shortint_default_add, the inner loop uses `add`). And since `mul_lsb` and `add` are default ops, then they output result that do not have any carries, so we aren't really testing the non-clean carry case

Reply:
Do you want to sync with @soonum on that to be sure everything is in order ?